compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -75,6 +75,31 @@ pub(crate) fn optimize_into_acir(
     time("SSA to ACIR", print_timings, || ssa.into_acir(&brillig, abi_distinctness))
 }
 
+/// Loop unrolling can return errors, since ACIR functions need to be fully unrolled.
+/// This meta-pass will keep trying to unroll loops and simplifying the SSA until no more errors are found.
+fn unroll_all_acir_loops(mut ssa: Ssa) -> Result<Ssa, RuntimeError> {
```

Comment:
Can you move this to `unroll.rs`?
I think the name is also too specific since we may still want to try unrolling some brillig loops in the future if they have constant bounds. Maybe something like `unroll_loops_iteratively`